src/OrchardCore.Modules/OrchardCore.Lucene/Services/LuceneQuerySource.cs

```diff
@@ -62,6 +62,8 @@ public async Task<object> ExecuteQueryAsync(Query query, IDictionary<string, obj
                     }
                 }
 
+                templateContext.AmbientValues.Add("TemplateType", "Query");
+
                 var tokenizedContent = await _liquidTemplateManager.RenderAsync(luceneQuery.Template, templateContext);
```

Comment:
You don't need the ambient value, add the encoder in this line, there is an overload that accepts it. This way you can let the Html one as the default.

Reply:
Oh really? My bad, I never saw that overload.